### DISCUSSION TITLE
Adding/using Arduino Library

### DISCUSSION THREAD

**Discussion by callimero** [SmingHub/Sming#2803] [Q&A] [answered]
Hej,

is there noob-compatible documentation how to use/include an Arduino Library for a own project?  I tried my was through and put the Lib here:

`Sming/Sming/Libraries/Temperature_LM75_Derived/`
(Temperature_LM75_Derived.cpp and  Temperature_LM75_Derived.h )

in component.mk:
`ARDUINO_LIBRARIES := Temperature_LM75_Derived `

```
cw@DESKTOP-PCKKJN6:~/I2C_Test$ make list-components |grep LM75
Temperature_LM75_Derived: /home/cw/smingwslser/Sming/Sming/Libraries/Temperature_LM75_Derived
```

make

```
Building /home/cw/smingwslser/Sming/Sming/out/Esp8266/debug/lib/clib-Temperature_LM75_Derived.a
make[2]: Nothing to be done for 'build'.

Building /home/cw/I2C_Test/out/Esp8266/debug/lib/clib-App-50bf66e1712e4b729216b688b66e4245.a
C+ /home/cw/I2C_Test/app/application.cpp
/home/cw/I2C_Test/app/application.cpp:32:10: fatal error: Temperature_LM75_Derived.h: No such file or directory
   32 | #include <Temperature_LM75_Derived.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

I can't exactly trace back what I have done to get to that point... A bit of a side project.

Best,
Carsten

**Comment by mikee47** [comment, marked as answer]
For that project you'll need to add a `component.mk` into the *library* with `COMPONENT_INCDIRS := src`.

See `https://sming.readthedocs.io/en/latest/information/develop/components.html#component-configuration`.

**Comment by callimero** [comment]
Thanks! So obvious now.
